src/code/client.coffee

```diff
@@ -229,6 +230,12 @@ class CloudFileManagerClient
             return @alert(err) if err
             @_fileOpened content, metadata, {openedContent: content.clone()}, @_getHashParams metadata
 
+  openUrlFile: (url) ->
+    urlProvider = new URLProvider()
```

Comment:
It's entirely a question of style. Since we only ever use this at most once (since we don't watch hash changes, to change the file we have to reload the app), and may not use it at all, it seemed to me to make sense to not instantiate it at all unless we need it.

But I can see it either way. Thoughts?